src/exporter/web_server.cc

```diff
@@ -0,0 +1,283 @@
+#include "web_server.h"
+#include "common/debug.h"
+#include "common/hostname.h"
+#include "global/global_init.h"
+#include "global/global_context.h"
+#include "exporter/DaemonMetricCollector.h"
+
+#include <boost/asio/ip/tcp.hpp>
+#include <boost/asio/ssl.hpp>   // SSL/TLS
+#include <boost/beast/core.hpp>
+#include <boost/beast/http.hpp>
+#include <boost/beast/version.hpp>
+#include <boost/thread/thread.hpp>
+#include <chrono>
+#include <cstdlib>
+#include <ctime>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <string>
+
+#define dout_context g_ceph_context
+#define dout_subsys ceph_subsys_ceph_exporter
+
+namespace beast = boost::beast;   // from <boost/beast.hpp>
+namespace http = beast::http;     // from <boost/beast/http.hpp>
+namespace net = boost::asio;      // from <boost/asio.hpp>
+namespace ssl = boost::asio::ssl; // from <boost/asio/ssl.hpp>
+using tcp = boost::asio::ip::tcp; // from <boost/asio/ip/tcp.hpp>
+
+// Base class for common functionality
+class web_connection {
+public:
+  virtual ~web_connection() = default;
+  virtual void start() = 0; // Pure virtual function to start the connection
+
+protected:
+  beast::flat_buffer buffer_{8192};
+  http::request<http::dynamic_body> request_;
+  http::response<http::string_body> response_;
+  net::steady_timer deadline_;
+
+  web_connection(net::any_io_executor executor, std::chrono::seconds timeout)
+      : deadline_(executor, timeout) {}
+
+    // Common request processing logic
+  void process_request() {
+    response_.version(request_.version());
+    response_.keep_alive(request_.keep_alive());
+
+    switch (request_.method()) {
+    case http::verb::get:
+      response_.result(http::status::ok);
+      create_response();
+      break;
+
+    default:
+      response_.result(http::status::method_not_allowed);
+      response_.set(http::field::content_type, "text/plain");
+      std::string body("Invalid request-method '" + std::string(request_.method_string()) + "'\n");
+      response_.body() = body;
+      break;
+    }
+    write_response();
+  }
+
+    // Construct a response message based on the request target
+  void create_response() {
+    if (request_.target() == "/") {
+      response_.set(http::field::content_type, "text/html; charset=utf-8");
+      std::string body("<html>\n"
+                        "<head><title>Ceph Exporter</title></head>\n"
+                        "<body>\n"
+                        "<h1>Ceph Exporter</h1>\n"
+                        "<p><a href='/metrics'>Metrics</a></p>"
+                        "</body>\n"
+                        "</html>\n");
+      response_.body() = body;
```

Comment:
Instead of providing a formatted HTML document linking to the /metrics endpoints, I'd rather return a plain-text or even empty doc with a 301 (Moved permanently) HTTP status.

Reply:
mmm... that it was exactly what we have previously. 
if we change that we can introduce some unexpected regression problem .... (very improbable, but possible ... who know what users have done interacting with ceph exporter)
Again.. I only have this concern about regression. but if nobodey shares my worries, nothing against and i will do the change.

Reply:
AFAIK from discussions with other colleagues nobody uses `ceph-exporter` when ceph is deployed in external mode mainly because it was not considered safe.

Reply:
Changed: 


Now... taking a look to the response :
```
# curl -v -k https://localhost:9926/
....
< HTTP/1.1 301 Moved Permanently
< Location: /metrics
< Content-Length: 0
< 
* Connection #0 to host localhost left intact
# 
```

apparently, no answer if we do not take a loot to the response fields:
```
# curl -k https://localhost:9926/
#
```

Reply:
Yes! Let's keep it as it used to be, I'd not personally like to change any patterns here & given we already have been following prometheus exporter, we should resemble the same here & expose html page